Automattic/jetpack
# Social: Add preview section to the social post modal

## Proposed changes:

* Extract some common logic to `useService` hook to easily get the service object
* Add preview section to the social post modal

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the editor preview feature - `define( 'JETPACK_SOCIAL_HAS_EDITOR_PREVIEW', true );`
* Goto post editor
* Open Jetpack or Social sidebar
* Under "Share this post" section, click on "Create custom posts"
* Confirm that you see the preview section in the modal
* Confirm that the custom message and attached media are reflected correctly.

<img width="1264" alt="image" src="https://github.com/user-attachments/assets/a2e43867-9b2e-452f-b1bc-155b2ff4849d">
